plugins/serialization/src/main/java/org/qbicc/plugin/serialization/InstanceInterceptor.java

```diff
@@ -0,0 +1,49 @@
+package org.qbicc.plugin.serialization;
+
+import static org.qbicc.graph.atomic.AccessModes.SingleUnshared;
+
+import org.qbicc.context.CompilationContext;
+import org.qbicc.interpreter.VmObject;
+import org.qbicc.type.definition.LoadedTypeDefinition;
+import org.qbicc.type.descriptor.ClassTypeDescriptor;
+
+/**
+ * This class enables VmObject instances to be modified
+ * immediately before they are serialized.
+ */
```

Comment:
What I had in mind was more along the lines of letting users declare some action in code, which we'd run via the interpreter somehow.  This does get us past the immediate issue, but I don't think we can expect to have compiler plugins handle every case in every library in the long term.

Reply:
It's pretty touchy to interpret code in LOWER.  In fact, I had originally written this code to do a `Vm.newInstance` using the default no-arg `FileDescriptor` constructor, but that didn't work.  (We happily tried to interpret it, but since it apparently hadn't been enqueued before, there was no schedule, and stepping through it in the debugger it happily interpreted the block-terminating `return` and never interpreted the store instructions that initialize `fd` and `handle` to `-1`).

Reply:
Yeah it would have to all happen in ADD, somehow.  Maybe we'd allow an instance method to be called which returns the value of the field to use on serialize; we'd call that right after instantiating the object and then store the value away somewhere. Then after ADD, we'd substitute all the values.